lib/lcevc/lcevc_dil.js

```diff
@@ -0,0 +1,281 @@
+        this.resizeObserver_ = new ResizeObserver((_entries) => {
+          window.requestAnimationFrame(() => {
+            this.playerResize_(this.lcevcContainer_, this.dil_,
+                this.media_, this.canvas_);
+          });
+        });
+
+        this.resizeObserver_.observe(this.parent_);
+      }
+    }
+  }
+
+  /**
+   * Resizing the container based on the source video dimensions.
+   * @private
+   */
+  playerResize_(lcevcContainer, lcevcDil, mediaElement, canvas) {
+    const lcevcContainerParent = /** @type {HTMLElement} */
+      (lcevcContainer.parentElement);
+
+    if (this.isSupported()) {
+      // We try to find the actual aspect ratio with the lcevc enhanced frame
+      // if this value is not a valid value we default it to 1920x1080
+      const aspectRatio = (lcevcDil && lcevcDil.aspectRatio > 0 ?
+        lcevcDil.aspectRatio :
+        mediaElement.videoWidth / mediaElement.videoHeight) || 1920 / 1080;
+      lcevcContainer.style.width =
+        `${(lcevcContainerParent).offsetWidth}px`;
+      if (lcevcDil && lcevcDil.isFullscreen) {
+        lcevcContainer.style.height =
+          `${window.innerHeight - this.videoControls_.offsetHeight}px`;
+
+        if (canvas.height < lcevcContainer.offsetHeight) {
+          this.canvas_.style.top =
+          `${(window.innerHeight - canvas.height) / 2}px`;
+        } else {
+          canvas.style.top = '';
+          canvas.style.height =
+            `${window.innerHeight - this.videoControls_.offsetHeight}px`;
+        }
+      } else {
+        lcevcContainer.style.height =
+          `${lcevcContainerParent.offsetWidth / aspectRatio}px`;
+
+        canvas.style.height = `${lcevcContainer.offsetHeight}px`;
+        canvas.style.top = '';
+      }
+    }
+  }
+
+  /**
+   * Set styles for video controls
+   * @private
+   * */
+  setControlsStyle_() {
+    if (this.videoControls_) {
+      this.videoControls_.style.position = 'absolute';
+      this.videoControls_.style.bottom = '0px';
+      this.videoControls_.style.width = '100%';
+      this.videoControls_.style.zIndex = '200';
+    }
+  }
+
+  /**
+   * Update Variant Level.
+   * @param {number} level
+   * @param {number} containerFormat
+   * @private
+   */
+  updateLevel_(level, containerFormat) {
+    this.createDil_();
+    if (this.dil_) {
+      this.level_ = level;
+      this.containerFormat_ = containerFormat;
+      this.dil_.setLevelSwitching(level, 1);
+      this.dil_.setContainerFormat(containerFormat);
+    }
+  }
+
+  /**
+   * Update current active variant
+   * @param {Array<shaka.extern.Track>} tracks
+   */
+  updateVariant(tracks) {
+    let containerFormat = 0;
+    for (let iterator = 0; iterator < tracks.length; iterator += 1) {
+      if (tracks[iterator].active) {
+        if (tracks[iterator].mimeType === 'video/webm') {
+          containerFormat = 1;
+        }
+        if (tracks[iterator].mimeType === 'video/mp4') {
+          containerFormat = 2;
```

Comment:
This is unclear.  Please create an enum (type `enum.<Number>`) to give these meaningful names.

Reply:
Added a new enum `shaka.lcevc.Dil.ContainerFormat` with changes in all the values referenced.